src/ctapipe/monitoring/calculator.py

```diff
@@ -0,0 +1,336 @@
+"""
+Definition of the ``PixelStatisticsCalculator`` class, providing all steps needed to
+calculate the montoring data for the camera calibration.
+"""
+
+import numpy as np
+from astropy.table import Table, vstack
+
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import (
+    ComponentName,
+    Dict,
+    Float,
+    Int,
+    List,
+    TelescopeParameter,
+    TraitError,
+)
+from ctapipe.monitoring.aggregator import StatisticsAggregator
+from ctapipe.monitoring.outlier import OutlierDetector
+
+__all__ = [
+    "PixelStatisticsCalculator",
+]
+
+
+class PixelStatisticsCalculator(TelescopeComponent):
+    """
+    Component to calculate statistics from calibration events.
+
+    The ``PixelStatisticsCalculator`` is responsible for calculating various statistics from
+    calibration events, such as pedestal and flat-field data. It aggregates statistics,
+    detects outliers, and handles faulty data periods.
+    This class holds two functions to conduct two different passes over the data with and without
+    overlapping aggregation chunks. The first pass is conducted with non-overlapping chunks,
+    while overlapping chunks can be set by the ``chunk_shift`` parameter for the second pass.
+    The second pass over the data is only conducted in regions of trouble with a high fraction
+    of faulty pixels exceeding the threshold ``faulty_pixels_fraction``.
+    """
+
+    stats_aggregator_type = TelescopeParameter(
+        trait=ComponentName(
+            StatisticsAggregator, default_value="SigmaClippingAggregator"
+        ),
+        default_value="SigmaClippingAggregator",
+        help="Name of the StatisticsAggregator subclass to be used.",
+    ).tag(config=True)
+
+    outlier_detector_list = List(
```

Comment:
I think for the configuration / provenance system to properly track the configuration of each of the detectors, there should be a `List(OutlierDetector).tag(config=True)` on the class. 

It can be empty and populated from this list of configuration dicts as you have it here. Just add the additional traitlet instead of using a plain list in `__init__`.

Reply:
This traitlet is currently a list of dicts since it holds besides the config for the `OutlierDetector` component also the column which the detector should be `apply_to`, i.e. on which column the `call` function of the component is applied to. Is this answer resolving the comment or did I misinterpret the comment?

Reply:
Yes, you didn't understand the point. I did not ask to change the current option.

I asked to add a second traitlet for the list of the **instances** of  `OutlierDetector`

Reply:
I see this is currently the same as in CameraCalibrator. I think we can leave it for now like this and after this is merged, I will have a look if this needs fixing in both.